reliability-engineering/terraform/modules/concourse-web/autoscaling-group.tf

```diff
@@ -7,6 +7,11 @@ resource "aws_autoscaling_group" "concourse_web" {
   target_group_arns   = [aws_lb_target_group.concourse_web.arn]
   load_balancers      = [aws_elb.concourse_web.name]
 
+  termination_policies = [
+    "OldestLaunchConfiguration",
+    "OldestInstance",
+  ]
```

Comment:
If a new instance fails to start, will ASG terminate this new instance first?

Reply:
yes it's possible that the deployment gets into a bad state in this case by killing the only healthy nodes ... however that's what staging is for :)

The checks for nodes becoming InService and (for workers) checks that they are registered with ATC should cause jobs to fail during staging, so catasophically bad deployments should be rare.

In any case like that, you are just back to the (existing) method of manually deploying terraform and kicking instances by hand.